File: common/src/main/java/com/namazustudios/socialengine/model/blockchain/CreateNeoTokenRequest.java
package com.namazustudios.socialengine.model.blockchain;

import com.namazustudios.socialengine.model.ValidationGroups;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;

import javax.validation.constraints.NotNull;
import java.util.List;
import java.util.Map;

@ApiModel(description = "Represents a request to create a NeoToken definition.")
public class CreateNeoTokenRequest {

    @NotNull(groups = {ValidationGroups.Create.class, ValidationGroups.Insert.class, ValidationGroups.Update.class})
    @ApiModelProperty("The token definition to deploy.")
    private Token token;

    @ApiModelProperty("Any meta data for this token.")
    private Map<String, Object> metaData;

    public Token getToken() {
        return token;
    }

    public void setToken(Token token) {
        this.token = token;
    }

    public Map<String, Object> getMetaData() {
        return metaData;
    }

    public void setMetaData(Map<String, Object> metaData) {
        this.metaData = metaData;
    }
}

File: web-ui-angular/elements-web-ui/src/app/api/models/blockchain/create-neo-token-request.ts
/* tslint:disable */

import { Token } from "./token";

/**
 * Represents a request to create a NeoToken definition.
 */
export interface CreateNeoTokenRequest {
  token: Token;

  /**
   * Is this token listed.
   */
  listed: boolean;

  /**
   * The elements contract id to mint this token with.
   */
  contractId: string;
}
